test/transform.js

```diff
@@ -470,3 +470,113 @@ check(
   , undefined
   , '1.' // pass '1.' as the prefix for toc entries
   )
+
+
+check(
+    [ '# My Module'
+    , 'Some text here'
+    , '## API'
+    , '### Method One'
+    , 'works like this'
+    , '### Method Two'
+    , '#### Main Usage'
+    , 'some main usage here'
+    ].join('\n')
+  , [ '**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*\n\n'
+    , '- [My Module](#my-module)\n'
+    ,   '  - [API](#api)\n'
+    ,     '    - [Method One](#method-one)\n'
+    ,     '    - [Method Two](#method-two)\n'
+    ,         '      - [Main Usage](#main-usage)\n\n\n'
+    ].join('')
+    , undefined
+    , undefined
+    , undefined
+    , undefined
+    , undefined
+    , undefined
+    , undefined
+    , 'md'
+)
+
+check(
+    [ '# My Module'
+    , 'Some text here'
+    , '## API'
+    , '### Method One'
+    , 'works like this'
+    , '### Method Two'
+    , '#### Main Usage'
+    , 'some main usage here'
+    ].join('\n')
+  , [ '**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*\n\n'
+    , '- [My Module](#my-module)\n'
+    ,   '  - [API](#api)\n'
+    ,     '    - [Method One](#method-one)\n'
+    ,     '    - [Method Two](#method-two)\n'
+    ,         '      - [Main Usage](#main-usage)\n\n\n'
+    ].join('')
+    , undefined
+    , undefined
+    , undefined
+    , undefined
+    , undefined
+    , undefined
+    , undefined
+    , 'mdx'
+)
+
+test('should use <!-- --> comments if syntax=md', function (t) {
+    var res = transform(
+                  [ '# My Module'
+                  , 'Some text here'
+                  , '## API'
+                  , '### Method One'
+                  , 'works like this'
+                  , '### Method Two'
+                  , '#### Main Usage'
+                  , 'some main usage here'
+                  ].join('\n')
+                  , undefined
+                  , undefined
+                  , undefined
+                  , undefined
+                  , undefined
+                  , undefined
+                  , undefined
+                  , 'md'
+              )
+
+    var lines = res.wrappedToc.split('\n')
+    var commentLines = lines.slice(0,2).concat(lines[lines.length-1])
```

Comment:
Pull out as a shared test helper, I think it's fine to leave it local to this file.